java-data/src/main/java/net/cryptic_game/backend/data/chat/ChannelAccess.java

```diff
@@ -0,0 +1,111 @@
+package net.cryptic_game.backend.data.chat;
+
+import com.google.gson.JsonObject;
+import net.cryptic_game.backend.base.api.client.ApiClient;
+import net.cryptic_game.backend.base.sql.models.TableModelAutoId;
+import net.cryptic_game.backend.base.utils.JsonBuilder;
+import net.cryptic_game.backend.data.user.User;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.annotations.Type;
+
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Entity
+@Table(name = "chat_channel_access")
+public class ChannelAccess extends TableModelAutoId {
+
+    @ManyToOne
+    @JoinColumn(name = "user_id", updatable = false, nullable = false)
+    @Type(type = "uuid-char")
+    private User user;
+
+    @ManyToOne
+    @JoinColumn(name = "channel_id", nullable = false, updatable = false)
+    @Type(type = "uuid-char")
+    private Channel channel;
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    public Channel getChannel() {
+        return channel;
+    }
+
+    public void setChannel(final Channel channel) {
+        this.channel = channel;
+    }
+
+    public static ChannelAccess join(final User user, final Channel channel, final Set<ApiClient> notifyUsers) {
+        Session sqlSession = sqlConnection.openSession();
+
+        ChannelAccess channelAccess = new ChannelAccess();
+        channelAccess.setUser(user);
+        channelAccess.setChannel(channel);
+
+        channelAccess.saveOrUpdate();
+        return channelAccess;
+    }
+
+    public static boolean leave(final User user, final Channel channel, Set<ApiClient> notifyUsers) {
+        try (Session sqlSession = sqlConnection.openSession()) {
+            sqlSession
+                    .createQuery("delete from ChannelAccess ca where ca.channel = :channel and ca.user = :user")
+                    .setParameter("channel", channel)
+                    .setParameter("user", user);
+            return true;
+        } catch (HibernateException e) {
+            final Logger log = LoggerFactory.getLogger(ChannelAccess.class);
```

Comment:
Define the Logger globaly as `private static final Logger log = LoggerFactory.getLogger(ChannelAccess.class);`